### PR TITLE
Database restore and invert BLAST

### PR DESCRIPTION
**Pull request by apetkau**
This makes a number of changes, and is meant to replace the previous merge request.

1. Implements `staramr db restore-default` which can be used to restore the default database (or build a database with the proper revisions initially).  This involves making a class [AMRDatabasesManager.py](https://github.com/phac-nml/staramr/pull/15/files#diff-5ca5d8d4cfd87dd998a6fa5c4c3a5d2a) to handle restoring the default database.
2. Inverts the direction of BLAST, so I'm BLASTing the AMR database files (amr genes) against the input genomes.  To do this, I must first run `makeblastdb` on all input files (I make symlinks to them first to not clutter up your input files directory).  This is done in [BlastHandler.py](https://github.com/phac-nml/staramr/pull/15/files#diff-99d67a7ecee1eb42b58e332bcb955a7bR105).  Most of the rest of the changes involves inverting the BLAST fields I access in [AMRHitHSP.py](https://github.com/phac-nml/staramr/pull/15/files#diff-17897e4b66fec3b247d5a00b9b22b7aa).
   1. Inverting BLAST means that running `staramr` against the latest ResFinder/PointFinder databases will work.  But, I found I get weird results since some genes aren't labelled properly in the fasta files.  So, I print out a `WARNING` if you are using the latest database. Done in [Search.py](https://github.com/phac-nml/staramr/pull/15/files#diff-8e9d37a44f3faed17b1c44ba3237eea1R143).
3. Since I no longer need to `makeblastdb` on the AMR gene files, I've removed this code when building the database.
4. I also no longer need to check for the `strand` when searching for point mutations because of the BLAST direction switch (there is a field for `sstrand` or subject strand, but nothing for `qstrand` or query strand available in the results, and the AMR gene string/genome contig string always appear in the same strand).  Removing strand checking is mostly done in [MutationPosition.py](https://github.com/phac-nml/staramr/pull/15/files#diff-7a503c4958d999a7939cad797167cccf) and subclasses.
   1. I do still need to check `sstrand` when it comes to making BLAST hit partitions though. This is where the changes in [BlastHitPartitions.py](https://github.com/phac-nml/staramr/pull/15/files#diff-30e80bf176a42a9a8c5f5f50b743843d) come from.
5. I changed to using an `OrderedDict` when getting database info since it makes it easier to check specific keys. Done in [AMRDatabaseHandler.py](https://github.com/phac-nml/staramr/pull/15/files#diff-1867e95b6fc2a935e6d9fc1b79c40f04R103).

### CODE REVIEW

File: staramr/databases/AMRDatabasesManager.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from staramr.databases.AMRDatabaseHandlerStripGitDir import AMRDatabaseHandlerStripGitDir
+
+logger = logging.getLogger('AMRDatabasesManager')
+
+"""
+A Class used to manage interactions with default and updatable ResFinder/PointFinder database installations.
+"""
+
+
+class AMRDatabasesManager:
```
Comment (peterk87) [inline staramr/databases/AMRDatabasesManager.py]: `Manager` is much more clear than `Factory` :+1:

File: staramr/blast/BlastHandler.py [hunk 2]
```diff
@@ -159,3 +189,8 @@ def _launch_blast(self, query, db, output):
         stdout, stderr = blastn_command()
         if stderr:
             raise Exception("error with [" + str(blastn_command) + "], stderr=" + stderr)
+
+    def _make_blast_db(self, path):
+        command = ['makeblastdb', '-in', path, '-dbtype', 'nucl', '-parse_seqids']
+        logger.debug(' '.join(command))
+        subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE).check_returncode()
```
[the remark this answers is not in the document]
Comment (apetkau) [reply staramr/blast/BlastHandler.py]: Thanks.  I didn't realize it was a recent addition.

File: staramr/Utils.py [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
+    :return: A string representation of the Dictionary.
     """
     s = ''
-    max_width = max([len(w[0]) for w in data])
+    max_width = max([len(k) for k in data])
 
-    for x, y in data:
-        d = dict(x=x.ljust(max_width), y=y)
-        s += '{x} = {y}\n'.format(**d)
+    for k in data:
```
[the remark this answers is not in the document]
Comment (apetkau) [reply staramr/Utils.py]: I think it's a fine one-liner and will make quite a nice addition - 645141ddae67e2f52143bb2322b6ffb0390bef4a

File: staramr/blast/results/AMRHitHSP.py [hunk 4]
```diff
@@ -154,8 +162,9 @@ def get_seq_record(self):
         Gets a SeqRecord for this hit.
         :return: A SeqRecord for this hit.
         """
-        return SeqRecord(Seq(self.get_genome_seq_in_amr_gene_strand()), id=self.get_amr_gene_id(),
-                         description=('isolate: {}, contig: {}, contig_start: {}, contig_end: {}, resistance_gene_start: {},'+
+        return SeqRecord(Seq(self.get_genome_seq()), id=self.get_amr_gene_id(),
+                         description=(
+                                     'isolate: {}, contig: {}, contig_start: {}, contig_end: {}, resistance_gene_start: {},' +
```
[the remark this answers is not in the document]
Comment (apetkau) [reply staramr/blast/results/AMRHitHSP.py]: Thanks. I didn't realize that a7979576002beb47f9ec2bac6e24c22711519291

**Pull request merged**